Formula/r/re2.rb

```diff
@@ -1,11 +1,10 @@
 class Re2 < Formula
   desc "Alternative to backtracking PCRE-style regular expression engines"
   homepage "https://github.com/google/re2"
-  url "https://github.com/google/re2/releases/download/2023-08-01/re2-2023-08-01.tar.gz"
-  version "20230801"
-  sha256 "d82d0efe2389949244445e7a6ac9a10fccc3d6a3d267ec4652991a51291647b0"
+  url "https://github.com/google/re2/archive/refs/tags/2023-09-01.tar.gz"
```

Comment:
Let's use the release artifact when possible.

```suggestion
  url "https://github.com/google/re2/releases/download/2023-09-01/re2-2023-09-01.tar.gz"
```